### PR TITLE
Userland: Port umount to LibMain

### PR DESCRIPTION
**Pull request by guerinoni**

### CODE REVIEW

File: Userland/Libraries/LibCore/System.cpp [hunk 1]
```diff
@@ -116,6 +116,18 @@ ErrorOr<void> mount(int source_fd, StringView target, StringView fs_type, int fl
...
 
+ErrorOr<void> umount(StringView mount_point)
+{
+    if (mount_point.is_null())
+        return Error::from_errno(EFAULT);
+
+    Syscall::SC_umount_params params {
+        { mount_point.characters_without_null_termination(), mount_point.length() },
+    };
+    int rc = syscall(SC_umount, &params);
```
Comment (IdanHo) [inline Userland/Libraries/LibCore/System.cpp:127]: huh? this syscall doesn't accept `SC_umount_params` it accepts a pointer and length, this won't work.

Comment (guerinoni) [reply Userland/Libraries/LibCore/System.cpp]: Thank you, now I understood better how syscalls works :)
Fixed

File: Userland/Libraries/LibCore/System.cpp [hunk 2]
```diff
@@ -116,6 +116,15 @@ ErrorOr<void> mount(int source_fd, StringView target, StringView fs_type, int fl
...
 
+ErrorOr<void> umount(StringView mount_point)
+{
+    if (mount_point.is_null())
+        return Error::from_errno(EFAULT);
+
+    int rc = syscall(SC_umount, mount_point.characters_without_null_termination(), mount_point.length());
+    HANDLE_SYSCALL_RETURN_VALUE("umount", rc, {});
+}
+
```
Comment (Rummskartoffel) [inline Userland/Libraries/LibCore/System.cpp:127]: This should probably go in a separate commit, I'd say.

Comment (guerinoni) [reply Userland/Libraries/LibCore/System.cpp:127]: @IdanHo what do you think? I will split it if needed :)

Comment (IdanHo) [reply Userland/Libraries/LibCore/System.cpp:127]: Sure, adding the LibCore wrapper is a separate atomic change that can stand on its own

Comment (guerinoni) [reply Userland/Libraries/LibCore/System.cpp:127]: Done :)

File: Userland/Utilities/CMakeLists.txt [hunk 3]
```diff
@@ -160,6 +160,7 @@ target_link_libraries(top LibMain)
 target_link_libraries(touch LibMain)
```
[the remark this answers is not in the document]
Comment (guerinoni) [reply Userland/Utilities/CMakeLists.txt:160]: Fixed, thanks :)

**Review by Rummskartoffel — APPROVED** [review]
LGTM :)

**Review by IdanHo — APPROVED** [review]
Cool, thanks
